tests/test_processing_stack.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import time
+
+from pytest import mark
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = os.environ.get("DEPLOY_ENV", "dev")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def detect_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+    logger.info("Detected Compute Environments: %s", compute_environment_detection_response)
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
```

Comment:
This can be simplified in the same way as `detect_state_machine` below.

Reply:
Same as above

Reply:
Done in 7f60bdd1c351d8dab432c00ec6830644a6707b2e